lint/linter.py

```diff
@@ -164,8 +164,13 @@ def __init__(cls, name, bases, attrs):
         if 'defaults' in attrs and attrs['defaults']:
             cls.map_args(attrs['defaults'])
 
-        if 'syntax' in attrs:
-            cls.register_linter(name)
+        if not cls.syntax and not cls.defaults['selectors']:
+            logger.error(
+                "{} disabled, either 'syntax' or 'selectors' must be specified"
```

Comment:
What about when a linter wants to lint all views (annotations)? I find the `*` special syntax to be an anti-pattern that we've been actively removing in the past and would suggest having neither a syntax nor selectors specification causes the linter to be active for all views.

Reply:
Can’t annotations specify the text and source scopes and thereby cover everything?

Reply:
Yeah, * is not needed anymore bc it meant selectors based anyway. annotations automatically gets enabled when there are comments in the buffer, or something like that.

Reply:
This check is heavily js influenced, and will just throw when selectors is not defined. Haha.

Reply:
Okay, different example: SublimeLinter-contrib-sublimesyntax. I need to check the view's file name (or its first line) to determine whether it is a syntax test file.

Reply:
These are yaml files.?!

cls.selectors is optional

Reply:
Syntax test files use whatever syntax they test for and could be any.

I'll wrap this up in a separate issue if I don't forget about it. This was just something I noticed while browsing the diff.